edxadapt/edxadapt.py

```diff
@@ -0,0 +1,172 @@
+import json
+import logging
+import pkg_resources
+
+from django.template import Template, Context
+from HTMLParser import HTMLParser
+from xblock.core import XBlock
+from xblock.fields import Scope, Boolean, Dict, List, String
+from xblock.fragment import Fragment
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from student.models import anonymous_id_for_user, User
+
+
+_ = lambda text: text
+log = logging.getLogger(__name__)
+html_parser = HTMLParser()
+
+
+class EdxAdaptXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    This XBlock automatically registers students in the EdxAdapt instance
+    using parameters configured by the teaching staff: params, skills, edx_adapt_api_url
+    """
+
+    has_author_view = True
+
+    display_name = String(
+        default='Adaptive Learning Enrollment',
+        scope=Scope.content,
+    )
+
+    params = Dict(
+        default={'pg': 0.25, 'ps': 0.25, 'pi': 0.1, 'pt': 0.5, 'threshold': 0.99},
+        display_name=_('BKT params'),
+        help=_('Parameters for Bayesian Knowledge Tracing (BKT) model'),
+        scope=Scope.content
+    )
+
+    skills = List(
+        default=[
+            'center', 'shape', 'spread', 'x axis', 'y axis', 'h to d',
+            'd to h', 'histogram', 'None'
+        ],
+        display_name=_('Course skills list'),
+        scope=Scope.content,
+        help=_(
+            'List of skills of this course. Each problem addresses certain skill. '
+            'Special skill "None" is used for those problems which belong to none.'
+        )
+    )
+    edx_adapt_api_url = String(
+        default='',
+        display_name=_('Edx Adapt API base URL'),
+        scope=Scope.content,
+        help=_('Edx Adapt API base URL, e.g. https://edx-adapt.example.com:443/api/v1')
+    )
+
+    edx_adapt_course_id = String(
```

Comment:
Thanks. I understand that/why you need it, I'm just not keen on the naming convention, since `course_id` means something very specific w/in edX.
That said, I can live with it for now.